src/main/java/io/gravitee/policy/aws/lambda/AwsLambdaPolicy.java

```diff
@@ -33,4 +51,90 @@ public AwsLambdaPolicy(AwsLambdaPolicyConfiguration configuration) {
+                }
+                return Completable.complete();
+            });
+    }
+
+    private <T extends HttpBaseExecutionContext> Single<InvokeResponse> invokeAndHandleLambda(T ctx) {
+        return Single
+            .fromFuture(invokeLambda(ctx.getTemplateEngine()))
+            .subscribeOn(Schedulers.io())
+            .flatMap(invokeResponse -> {
+                log.debug("AWS Lambda function has been invoked successfully");
+                if ("Handled".equals(invokeResponse.functionError()) || "Unhandled".equals(invokeResponse.functionError())) {
+                    return processError(ctx, null, AwsLambdaError.AWS_LAMBDA_INVALID_RESPONSE);
+                } else if (invokeResponse.statusCode() >= 200 && invokeResponse.statusCode() < 300) {
+                    return processSuccess(ctx, invokeResponse);
+                }
+                return processError(ctx, null, AWS_LAMBDA_INVALID_STATUS_CODE);
+            })
+            .onErrorResumeNext(throwable -> {
+                log.error("An error occurs while invoking lambda function", throwable);
+                return processError(ctx, throwable, AwsLambdaError.AWS_LAMBDA_INVALID_RESPONSE);
+            });
+    }
+
+    private <T extends HttpBaseExecutionContext> Single<InvokeResponse> processError(T ctx, Throwable throwable, AwsLambdaError error) {
+        ExecutionFailure failure = new ExecutionFailure(error.getStatusCode())
+            .key(error.name())
+            .message(throwable != null ? error.getMessageWithDetails(error, throwable) : error.getMessage());
+
+        return switch (ctx) {
+            case HttpMessageExecutionContext messageCtx -> messageCtx
+                .interruptMessagesWith(failure)
+                .ignoreElements()
+                .<InvokeResponse>toMaybe()
+                .toSingle();
+            case HttpPlainExecutionContext plainCtx -> plainCtx.interruptWith(failure).<InvokeResponse>toMaybe().toSingle();
+            default -> Single.error(new IllegalArgumentException("Unsupported context type"));
+        };
+    }
+
+    private <T extends HttpBaseExecutionContext> Single<InvokeResponse> processSuccess(T ctx, InvokeResponse result) {
+        var tplEngine = ctx.getTemplateEngine();
+        tplEngine.getTemplateContext().setVariable(TEMPLATE_VARIABLE, new LambdaResponse(result));
+
+        if (configuration.getVariables() != null) {
+            configuration
+                .getVariables()
+                .forEach(variable -> {
+                    try {
+                        String extValue = (variable.getValue() != null) ? tplEngine.evalNow(variable.getValue(), String.class) : null;
+                        ctx.setAttribute(variable.getName(), extValue);
+                    } catch (Exception exception) {
+                        log.error("An error occurs while setting variable {}", variable.getName(), exception);
```

Comment:
If the evaluation fails, maybe it's better to make the request fail also by propagating the error otherwise, there are chances the next policy expecting this attribute will fail too, and it will be harder to debug. WDYT?